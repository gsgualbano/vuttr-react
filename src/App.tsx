import React from 'react';
import { Provider } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import { ThemeProvider } from 'styled-components';

import useDarkMode from 'use-dark-mode';

import MainPage from './pages/Main';
import GlobalStyles from '~/styles/global';

import store from './store';
import 'react-toastify/dist/ReactToastify.css';


import lightTheme from './themes/light';
import darkTheme from './themes/dark';

const App: React.FC = (): JSX.Element => {
  const darkMode = useDarkMode();

  return (
    <ThemeProvider theme={darkMode.value ? darkTheme : lightTheme}>
      <Provider store={store}>
        <GlobalStyles />
        <MainPage />
        <ToastContainer />
      </Provider>
    </ThemeProvider>
  );
};

export default App;
