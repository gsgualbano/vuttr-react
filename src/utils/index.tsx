/* eslint-disable max-len */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { reducer } from '../store/ducks/tools';

interface Theme {
  colors: {
    background: string;
    textColor: string;
    cardBackground: string;
    cardBorderColor: string;
    modalBackground: string;
    modalBorderColor: string;
    linkColor: string;
    linkVisitedColor: string;
    buttonOutlineBackground: string;
    InputBackground: string;
    InputBorderColor: string;
  };
}

/** redux mock to render components that depends on the redux */
export const renderWithRedux = (
  component: JSX.Element,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(<Provider store={store}>{component}</Provider>),
  store,
});

/** render components thats depends on the theme provider */
export const renderWithTheme = (component: JSX.Element, theme: Theme) => render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

/** render components thats depends on the redux and theme provider */
export const renderWithReduxAndTheme = (
  component: JSX.Element,
  theme: Theme,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>{component}</Provider>
    </ThemeProvider>,
  ),
  store,
});
