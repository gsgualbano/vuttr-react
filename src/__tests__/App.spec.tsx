import React from 'react';
import { renderWithRedux } from '~/utils';
import { cleanup, fireEvent } from '@testing-library/react';

import App from '~/App';

import 'jest-styled-components';

afterEach(cleanup);

describe('<App />', () => {
  it('render correctly', () => {
    const { asFragment } = renderWithRedux(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('To switch to dark mode', () => {
    const { getByText, getByTestId } = renderWithRedux(<App />);
    fireEvent.click(getByText('Dark Theme'));
    expect(getByTestId('main-page')).toHaveStyleRule('background', '#1B1A1A');
  });
});
