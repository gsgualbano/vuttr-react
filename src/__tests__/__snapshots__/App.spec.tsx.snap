// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render correctly 1`] = `
<DocumentFragment>
  .c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 16px;
  color: #ffffff;
  background: #365df0;
  padding: 5px 10px;
  border-radius: 5px;
  border: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11:hover {
  background: #2f55cc;
}

.c11:click {
  background: #244aa8;
  background-color: #244aa8;
}

.c3 {
  font-weight: 600;
  font-size: 42px;
  margin: 30px 0;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.c6 {
  font-weight: 600;
  font-size: 30px;
  margin: 30px 0;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 10px 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 5px;
  border: 1px solid;
  border-color: #ebeaed;
  background-color: #f5f4f6;
  width: 200pt;
  margin-right: 10px;
  padding: 10px;
  font-family: 'Source Sans Pro',sans-serif;
}

.c10 {
  border: none;
  width: 100%;
  background: inherit;
  outline: none;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 {
  width: 14px;
  height: 14px;
  margin-right: 5px;
}

.c13 g > circle.a {
  stroke: #ffffff;
  stroke-width: 2px;
}

.c13 g > path.b {
  stroke: #ffffff;
  stroke-width: 7px;
}

.c9 {
  width: 14px;
  height: 14px;
  margin-right: 5px;
}

.c9 g > path.a {
  stroke: #b1adb9;
}

.c9 g > circle.a {
  stroke: #b1adb9;
}

.c4 {
  max-height: 30px;
  background: transparent;
  border: none;
  color: #b1adb9;
  font-size: 20px;
  margin: auto 0;
}

.c5 {
  width: 18px;
  height: 18px;
  margin-right: 7px;
}

.c5 g > path.a {
  stroke: #b1adb9;
}

.c5 g > circle.a {
  stroke: #b1adb9;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c15 {
  margin-top: 20px;
  color: gray;
  text-align: center;
}

.c1 {
  width: 100%;
  padding: 0 10px;
}

.c0 {
  background: #fcfcfd;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c11 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c7 {
    margin-bottom: 10px;
    -webkit-box-pack: none;
    -webkit-justify-content: none;
    -ms-flex-pack: none;
    justify-content: none;
  }
}

@media (max-width:767px) {
  .c8 {
    width: 100%;
  }
}

@media (min-width:576px) {
  .c1 {
    max-width: 540px;
  }
}

@media (min-width:768px) {
  .c1 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 960px;
  }
}

<div
    class="c0"
    data-testid="main-page"
  >
    <div
      class="c1"
    >
      <div
        class=""
      >
        <div
          class="c2"
        >
          <h1
            class="c3"
          >
            VUTTR
          </h1>
          <button
            class="c4"
            type="button"
          >
            <svg
              class="c5"
            >
              Icon-Dark Mode 2px.svg
            </svg>
            Dark Theme
          </button>
        </div>
        <h2
          class="c6"
        >
          Very Useful Tools to Remember
        </h2>
        <div
          class="c2"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <svg
                class="c9"
              >
                Icon-Search-2px.svg
              </svg>
              <input
                class="c10"
                placeholder="Search"
                type="text"
              />
            </div>
            <input
              checked=""
              class=""
              type="checkbox"
            />
            search in tags only
          </div>
          <button
            class="c11"
            data-testid="primary-button"
            type="button"
          >
            <div
              class="c12"
            >
              <svg
                class="c13"
              >
                Icon-Plus-Circle-2px.svg
              </svg>
            </div>
            Add
          </button>
        </div>
      </div>
      <div
        class="c14"
      >
        <p
          class="c15"
        >
          No tools yet, to add a new tool press add!
        </p>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
